src/main/groovy/org/ajoberstar/gradle/git/ghpages/GithubPagesPlugin.groovy

```diff
@@ -56,28 +59,49 @@ class GithubPagesPlugin implements Plugin<Project> {
 			with extension.pages.realSpec
 			into { extension.workingDir }
 			doFirst {
-				extension.workingDir.deleteDir()
-				ext.repo = Grgit.clone(
-						uri: extension.repoUri,
-						refToCheckout: extension.targetBranch,
-						dir: extension.workingDir,
-						credentials: extension.credentials?.toGrgit()
-				)
+				def repo = null
+				try {
+					// attempt to reuse existing repository
+					repo = Grgit.open(dir: extension.workingDir)
+					if (extension.repoUri == repo.remote.list().find { it.name == 'origin' }?.url &&
+							repo.branch.current.name == extension.targetBranch) {
+						repo.fetch()
+						repo.reset(commit: 'origin/' + extension.targetBranch, mode: ResetOp.Mode.HARD)
```

Comment:
We should probably also run [`repo.clean(directories: true, ignore: false)`](http://ajoberstar.org/grgit/docs/groovydoc/org/ajoberstar/grgit/operation/CleanOp.html) to get rid of any untracked files.